apps/web/src/components/WalletConnect/WcProvider.tsx

```diff
@@ -0,0 +1,82 @@
+import { type WalletKitTypes } from "@reown/walletkit";
+import { useDynamicModalContext } from "@umami/components";
+import { createWalletKit, useCurrentAccount, walletKit } from "@umami/state";
+import { formatJsonRpcError } from "@walletconnect/jsonrpc-utils";
+import { getSdkError, parseUri } from "@walletconnect/utils";
+import { type PropsWithChildren, useEffect } from "react";
+
+import SessionProposalModal from "./SessionProposalModal";
+
+export const context = {};
+
+export const WalletConnectProvider = ({ children }: PropsWithChildren) => {
+  const currentAccount = useCurrentAccount();
+  const { openWith } = useDynamicModalContext();
+
+  useEffect(() => {
+    const initializeWallet = async () => {
+      try {
+        await createWalletKit();
+        walletKit.on("session_proposal", proposal => void onSessionProposal(proposal));
+        walletKit.on("session_request", request => void onSessionRequest(request));
+      } catch (error) {
+        console.error("Error initializing Web3Wallet:", error);
+      }
+    };
+
+    void initializeWallet();
+  });
+
+  async function onSessionProposal(proposal: WalletKitTypes.SessionProposal) {
+    console.log("Session proposal received", proposal);
+    if (!currentAccount) {
+      console.error("No current account found");
+      return;
+    }
+
+    const modal = <SessionProposalModal proposal={proposal} />;
+
+    try {
+      await openWith(modal, {});
+    } catch (error) {
+      console.error("Failed to open SessionProposalModal modal:", error);
+    }
+
+    return () => {};
+  }
+
+  async function onSessionRequest(event: WalletKitTypes.SessionRequest) {
```

Comment:
We use arrow functions syntax instead of function declarations:
`const onSessionRequest = (event: WalletKitTypes.SessionRequest) => {...}`

Please update here and in other places in order for the code to have the same style across the codebase